contactql/visitor.go

```diff
@@ -69,7 +80,21 @@ func (v *Visitor) VisitCondition(ctx *gen.ConditionContext) interface{} {
 		comparator = resolvedAlias
 	}
 
-	return &Condition{key: key, comparator: comparator, value: value}
+	var propType PropertyType
+
+	if attributeKeys[propKey] {
+		propType = PropertyTypeAttribute
+	} else if urns.IsValidScheme(propKey) {
+		propType = PropertyTypeScheme
+
+		if v.redaction == envs.RedactionPolicyURNs {
+			v.errors = append(v.errors, errors.New("URN scheme not allowed"))
```

Comment:
maybe `cannot query on redacted URNs` just to make it super obvious as to WHY this is an error.